hack/etcd_cleanup.sh

```diff
@@ -0,0 +1,68 @@
+#!/bin/bash
+
+# Copyright 2014 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This script is designed to run on master and assume etcd is running on localhost.
+# It would recursively look into etcd keys and remove any empty directory
+# It will download it's own version of etcdctl.
+# Usage: (on master the mount point may not have exec flag, use bash to run the script)
+# > bash etcd_cleanup.sh
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+TMPDIR=$(mktemp -d)
+echo "using temp directory $TMPDIR"
+MLINE=$( (MYDIR=$TMPDIR; until findmnt $MYDIR ; do MYDIR=$(dirname $MYDIR) ; done) | tail -n 1 | tr -s ' ')
+FSTRING=$(echo $MLINE | cut -f 4 -d ' ')
+MPOINT=$(echo $MLINE | cut -f 1 -d ' ')
+
+if [[ $FSTRING == *"noexec"* ]]; then
+  echo "non-executable mount point $MPOINT detected. remounting with exec flag..."
+  sudo mount -o remount,exec $MPOINT
+  TMP_REMOUNTED=1
+else
+  TMP_REMOUNTED=0
+fi
+
+function get_etcdctl {
+	curl -s -L  https://github.com/coreos/etcd/releases/download/v2.3.4/etcd-v2.3.4-linux-amd64.tar.gz -o $1/etcd.tar.gz > /dev/null
```

Comment:
This script is meant to stand alone, dependencies are not good.

However I would suggest to add a Dockerfile and build this script and the
etcdctl binary into a single container, so that only one thing needs to be
distributed. (Did I already say this? Can't remember)

On Mon, Aug 8, 2016 at 2:24 PM, Jeff Grafton <notifications@github.com>
wrote:

> In hack/etcd_cleanup.sh
> <https://github.com/kubernetes/kubernetes/pull/30004#discussion_r73957953>
> :
>
> > +TMPDIR=$(mktemp -d)
> > +echo "using temp directory $TMPDIR"
> > +MLINE=$( (MYDIR=$TMPDIR; until findmnt $MYDIR ; do MYDIR=$(dirname $MYDIR) ; done) | tail -n 1 | tr -s ' ')
> > +FSTRING=$(echo $MLINE | cut -f 4 -d ' ')
> > +MPOINT=$(echo $MLINE | cut -f 1 -d ' ')
> > +
> > +if [[ $FSTRING == *"noexec"* ]]; then
> > +  echo "non-executable mount point $MPOINT detected. remounting with exec flag..."
> > +  sudo mount -o remount,exec $MPOINT
> > +  TMP_REMOUNTED=1
> > +else
> > +  TMP_REMOUNTED=0
> > +fi
> > +
> > +function get_etcdctl {
> > +	curl -s -L  https://github.com/coreos/etcd/releases/download/v2.3.4/etcd-v2.3.4-linux-amd64.tar.gz -o $1/etcd.tar.gz > /dev/null
>
> any chance you can use the code in hack/lib/etcd.sh to do this instead?
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/30004/files/553961f2d5f05e2a097123c31f51644933896c3a#r73957953>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAngljayhxn5CdJqOZ4sqVYbrcdwsQTKks5qd56pgaJpZM4Jb5Ni>
> .
>